### PR TITLE
Circle CI to build Android dependencies and rnkiwimobile

### LINKED ISSUE #740: More automation for Android integration
- [x] ~~Investigate if we can get rid of intermediate snapshots https://github.com/kiwicom/mobile/blob/master/RNAndroidPlayground/rnkiwimobile/build.gradle#L48-L50 and take them from `node_modules` now that is in the same directory~~ <-- we won't do it
- [x] Create a script to create SNAPSHOT(s) of `rnkiwimobile` and other dependencies
- [x] Remove versions from dependencies.gradle and rely on package.json
- [x] Make CircleCI to create SNAPSHOT(s) of `rnkiwimobile` and other dependencies

### PR DESCRIPTION
**Pull request by ferrannp**
Fixes #740.

- Generates dependencies SNAPSHOTS(s) taking version from package.json if they do not exist
- Generates JS and `rnkiwmobile` library taking version from package.json if it does note xist

### CODE REVIEW

File: scripts/buildAndroidSnapshots.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+const DEFAULT_URL =
+  'http://trinerdis.cz:8000/repository/snapshots/com/trinerdis/skypicker/';
+
+if (!process.env.ANDROID_DEPLOYMENT_PASSWORD) {
+  throw Error('You need to pass ANDROID_DEPLOYMENT_PASSWORD.');
+}
+
+// Read native dependencies versions from package.json (maps, vector icons...)
+const buildPackage = require('../.build/package.json');
```
Comment (mrtnzlml) [inline scripts/buildAndroidSnapshots.js]: This file is going to be eventually removed. Is there a better solution? (maybe just the `packing.json` from the root)

File: scripts/buildAndroidSnapshots.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+// Get rnkiwimobile version
+const RNKiwiMobileVersion = buildPackage.version;
+
+const deployToTrinerdis = (packageName, manualVersion, buildJS = false) => {
+  const version = manualVersion
+    ? manualVersion
+    : getDependencyVersion(packageName);
+
+  return new Promise((resolve, reject) =>
+    urlExists(
```
Comment (mrtnzlml) [inline scripts/buildAndroidSnapshots.js]: Just a question - why is this necessary?

File: scripts/buildAndroidSnapshots.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    ),
+  );
+};
+
+(() => {
+  console.log(`Start building Android SNAPSHOT(s)...`);
+  Promise.all([
+    deployToTrinerdis('react-native-maps'),
+    deployToTrinerdis('react-native-vector-icons'),
+    deployToTrinerdis('react-native-tooltips'),
```
Comment (mrtnzlml) [inline scripts/buildAndroidSnapshots.js]: Ah. Uh. So we didn't solve the pain with native deps... 😞

File: scripts/buildAndroidSnapshots.js (hunk 1 above)
Comment (ferrannp) [reply scripts/buildAndroidSnapshots.js]: Mmmm, if we cannot use that file, we will need to check several ones because you have dependencies spread. Like `react-native` in the root but others are in shared. Want me to change that?

File: scripts/buildAndroidSnapshots.js (hunk 2 above)
Comment (ferrannp) [reply scripts/buildAndroidSnapshots.js]: So I talk this with @Blackhex and we would collapse trinerdis if we deploy every time we merge so I am checking if that url exists and just deploying when we change versions.

File: scripts/buildAndroidSnapshots.js (hunk 3 above)
Comment (ferrannp) [reply scripts/buildAndroidSnapshots.js]: Also talked with @Blackhex and we need those for Android as they don't really exist out there deployed. We need the SNAPSHOTS(s) in order to avoid build time. Updating a native dependency should not require any work, adding a new one yes.

File: scripts/buildAndroidSnapshots.js (hunk 1 above)
Comment (mrtnzlml) [reply scripts/buildAndroidSnapshots.js]: We went from this:

1. add native dep
2. add native dep to the package.json

To this:

1. add native dep
2. add native dep to the package.json
3. add native dep to the buildAndroidSnapshots.js

Instead of this:

1. add native dep

That's what I don't like about it - it's error prone.

Comment (ferrannp) [reply scripts/buildAndroidSnapshots.js]: Mmm that is not entirely true because when you added a native dep, someone had to go here https://gitlab.skypicker.com/mobile/react-native-hotels and it + create the Snapshot process.

Now that we have it though all in one repo, maybe we can think of way to reduce steps.
